grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingOptionsBuilder.java

```diff
@@ -95,7 +100,17 @@ public HttpJsonTranscodingOptions build() {
         if (queryParamMatchRules == null) {
             matchRules = DEFAULT_QUERY_PARAM_MATCH_RULES;
         } else {
-            matchRules = Sets.immutableEnumSet(queryParamMatchRules);
+            if (queryParamMatchRules.size() == 1 && queryParamMatchRules.contains(JSON_NAME)) {
+                // If neither LOWER_CAMEL_CASE nor ORIGINAL_FIELD is set, add ORIGINAL_FIELD by default.
```

Comment:
Optional) This rule looks like it can go stale easily if a new type is added.

```suggestion
            checkArgument(queryParamMatchRules.contains(LOWER_CAMEL_CASE) || queryParamMatchRules.contains(ORIGINAL_FIELD));
```

Reply:
@ikhoon Would you mind checking this comment? 😉

Reply:
> ```java
> checkArgument(queryParamMatchRules.contains(LOWER_CAMEL_CASE) ||
>     queryParamMatchRules.contains(ORIGINAL_FIELD));
> ```

This condition prevents `JSON_NAME` from being used alone in `HttpJsonTranscodingQueryParamMatchRule`. As a result, users would have to handle different cases differently within the same enum.

While `.queryParamMatchRules(ORIGINAL_FIELD)` works fine, triggering an error for `.queryParamMatchRules(JSON_NAME)` might be confusing. Since the Javadoc states `ORIGINAL_FIELD` is considered the default, it seems better to use `ORIGINAL_FIELD` as a fallback when `JSON_NAME` does not match.

I will consider a better approach such as introducing an interface if we need more flexible matching rules.